browser/main/lib/dataApi/attachmentManagement.js

```diff
@@ -342,6 +342,54 @@ function getAbsolutePathsOfAttachmentsInContent (markdownContent, storagePath) {
   return result
 }
 
+/**
+ * @description Copies the attachments to the storage folder and returns the mardown content it should be replaced with
+ * @param {String} markDownContent content in which the attachment paths should be found
+ * @param {String} filepath The path of the file with attachments to import
+ * @param {String} storageKey Storage key of the destination storage
+ * @param {String} noteKey Key of the current note. Will be used as subfolder in :storage
+ */
+function importAttachments (markDownContent, filepath, storageKey, noteKey) {
+  return new Promise((resolve, reject) => {
+    const nameRegex = /(!\[.*?]\()(.+?\..+?)(\))/g
+    let attachName = nameRegex.exec(markDownContent)
+    const promiseArray = []
+    const beginPath = filepath.match(/\/.+\//)[0]
```

Comment:
Can you use `path.dirname` for this? It's much faster than using Regex
![image](https://user-images.githubusercontent.com/12984316/50053959-49a41b00-016f-11e9-95ad-4caf8e9b0805.png)